it/controllers/agentAppointment/ConfirmAgentAppointControllerFSOnISpec.scala

```diff
@@ -18,110 +18,365 @@ import java.util.UUID
-    implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId(testSessionId)))
+      s"has a header of '$headingText' in welsh" in {
+        document.select(headingSelector).text shouldBe headingTextWelsh
+      }
 
-    val account = groupAccount(true)
-    val agentCode = 1001
-    val agentName = "Test Agent"
-    val backLinkUrl = "some/url"
+      s"doesn't have the text on the screen for the assigned properties" in {
+        document.select(assignedToPropertiesSelector).size() shouldBe 0
+      }
+
+      s"has text on the screen of $thisAgentCanText in welsh" in {
+        document.select(thisAgentCanSelector).text shouldBe thisAgentCanTextWelsh
+      }
+
+      s"has a bullet point on the screen of $addPropertiesText in welsh" in {
+        document.select(addPropertiesSelector).text shouldBe addPropertiesTextWelsh
+      }
+
+      s"has a bullet point on the screen of $ratingsList2017TextWelsh in welsh" in {
+        document.select(ratingsListSelector).text shouldBe ratingsList2017TextWelsh
+      }
+
+      s"has a subheading on the screen of $whatHappensNextText in welsh" in {
+        document.select(whatHappensNextSelector).text shouldBe whatHappensNextTextWelsh
+      }
+
+      s"has text on the screen of $youCanAssignText in welsh" in {
+        document.select(youCanAssignSelector).text shouldBe youCanAssignTextWelsh
+      }
+
+      s"has a $managingAgentsText link which takes you to Manage agent properties screen in welsh" in {
+        document.select(managingAgentsLinkSelector).text() shouldBe managingAgentsTextWelsh
+        document.select(managingAgentsLinkSelector).attr("href") shouldBe managingAgentsLinkHref
+      }
+
+      s"has a $goToHomeText link which takes you to the home screen in welsh" in {
+        document.select(goToHomeSelector).text() shouldBe goToHomeTextWelsh
+        document.select(goToHomeSelector).attr("href") shouldBe goToHomeLinkHref
+      }
+
+    }
+
+  }
+
+  def getDocument(language: Language, ratingsList: String): Document = {
+
+    val (bothRatingsListChoice, specificRatingsListChoice): (Option[Boolean], Option[String]) = ratingsList match {
+      case "Both" => (Some(true), None)
+      case "2023" => (Some(false), Some("2023"))
+      case "2017" => (Some(false), Some("2017"))
+      case _ => (None, None)
+    }
 
     val managingPropertyData: ManagingProperty = ManagingProperty(
       agentCode = agentCode,
-      agentOrganisationName = "Some Org",
+      agentOrganisationName = agentName,
       isCorrectAgent = true,
-      managingPropertyChoice = All.name,
+      managingPropertyChoice = ChooseFromList.name,
       agentAddress = "An Address",
       backLink = None,
       totalPropertySelectionSize = 2,
       propertySelectedSize = 2,
-      bothRatingLists = bothRatingList,
-      specificRatingList = specificRatingList
+      singleProperty = true,
+      bothRatingLists = bothRatingsListChoice,
+      specificRatingList = specificRatingsListChoice
```

Comment:
All the tests are when the choice is ChooseFromList & assigning all 2 properties to the agent. I know the content shouldn't change, but we run this test every time
```
s"doesn't have the text on the screen for the assigned properties" in {
        document.select(assignedToPropertiesSelector).size() shouldBe 0
}
```

It might be worth testing assign to all/some/one/none to make sure the different variations of assignedPropertiesText don't show up. I don't think we need separate tests, adding it to the existing is fine e.g all & 2023, some & 2017, one property & both list years then one additional test for assigned to none

Reply:
Okay cool, that's fine then 👍